### PR TITLE
FileSystem: Use _stat64 routines

### PR DESCRIPTION
**Pull request by xantares**
allows compilation with gcc/mingw

### CODE REVIEW

File: common/GL/ContextWGL.cpp [hunk 1]
```diff
@@ -241,7 +241,7 @@ namespace GL
 			}
 
 			// re-init glad-wgl
-			if (!gladLoadWGLLoader([](const char* name) -> void* { return wglGetProcAddress(name); }, m_dc))
+			if (!gladLoadWGLLoader([](const char* name) -> void* { return (void*)wglGetProcAddress(name); }, m_dc))
```
Comment (stenzek) [inline common/GL/ContextWGL.cpp:244]: `if (!gladLoadWGLLoader([](const char* name) -> { return reinterpret_cast<void*>(wglGetProcAddress(name)); }, m_dc))`

File: common/GL/ContextWGL.cpp [hunk 2]
```diff
@@ -319,7 +319,7 @@ namespace GL
 			}
 
 			// re-init glad-wgl
-			if (make_current && !gladLoadWGLLoader([](const char* name) -> void* { return wglGetProcAddress(name); }, m_dc))
+			if (make_current && !gladLoadWGLLoader([](const char* name) -> void* { return (void*)wglGetProcAddress(name); }, m_dc))
```
Comment (stenzek) [inline common/GL/ContextWGL.cpp:322]: `if (make_current && !gladLoadWGLLoader([](const char* name) -> { return reinterpret_cast<void*>(wglGetProcAddress(name)); }, m_dc))`

File: common/Windows/WinHostSys.cpp [hunk 3]
```diff
@@ -47,6 +48,16 @@ long __stdcall SysPageFaultExceptionFilter(EXCEPTION_POINTERS* eps)
 	{
 		return EXCEPTION_CONTINUE_SEARCH;
 	}
+#else
+  try
+  {
+    return DoSysPageFaultExceptionFilter(eps);
+  }
+  catch (...)
```
Comment (stenzek) [inline common/Windows/WinHostSys.cpp:56]: This isn't going to work, unless your compiler transforms SEH exceptions into C++ exceptions. As far as I'm aware, MSVC is the only one which does this, but I could be wrong (although it's not really something you want, and it's disabled by default).

Fortunately, there's other options, like using vectored exception handlers, which we use on 64-bit anyway.

Comment (xantares) [reply common/Windows/WinHostSys.cpp:56]: afaik gcc handles c++ exception as seh or sjlj

**Comment by lightningterror** [comment]
@xantares Status?

**Comment by xantares** [comment]
@lightningterror this is still needed

**Review by lightningterror — COMMENTED** [review]
It was discussed that the try/catch bits are undesired and we also use VEH anyway.
Should also be properly formatted.

**Comment by lightningterror** [comment]
Closing for the reasons stated before.

**Comment by xantares** [comment]
@lightningterror what if I drop that commit, the rest should be good

**Comment by stenzek** [comment]
I don't know why we would want to build with mingw anyway. There's plenty of issues with headers being out of date, subtle ABI differences, and it provides no advantage.

I have a branch with clang-cl support, which if you want to use non-MSVC codegen, works much better.

**Pull request closed without merging**